ft-sdk/src/session.rs

```diff
@@ -49,13 +49,64 @@ impl SessionID {
         self.update_one(conn, Some(user_id), None)
     }
 
-    /// Set the data for the given session.
-    pub fn set_data(
+    /// Get the session data object.
+    /// Useful for fetching the entire session data in a single db call. Use
+    /// [get_key](SessionID::get_key) instead if you only need a single key
+    pub fn data(
         &self,
         conn: &mut ft_sdk::Connection,
-        data: serde_json::Value,
-    ) -> Result<SessionID, ft_sdk::Error> {
-        self.update_one(conn, None, Some(data))
+    ) -> Result<serde_json::Value, diesel::result::Error> {
+        use diesel::prelude::*;
+        use ft_sdk::schema::fastn_session;
+
+        let data = fastn_session::table
+            .select(fastn_session::data)
+            .filter(fastn_session::id.eq(self.0.as_str()))
+            .first::<String>(conn)?;
+
+        Ok(serde_json::from_str(&data).expect("session data must be serializable json object"))
+    }
+
+    pub fn get_key<S: AsRef<str>>(
+        &self,
+        conn: &mut ft_sdk::Connection,
+        k: S,
+    ) -> Result<serde_json::Value, GetKeyError> {
```

Comment:
Make it `get_key<S: AsRef<str>, V: serde::de::DeserializedOwned>() -> Result<T, GetKeyError>`.

Reply:
Thanks. done in https://github.com/FifthTry/ft-sdk/pull/52/commits/5c878f5187d5a44fc02833adac719e9ba45895ab